src/v2/api/services/PermissionService.ts

```diff
@@ -48,10 +47,24 @@ export class PermissionService {
     return true;
   }
 
-  checkPermission (roles: DbRole[], permission: string) {
+  async checkParentIdPermission (roleId: string, permission: string): Promise<boolean> {
+    const role = await this.roleRepository.findById(roleId);
+    return this.checkRolePermission(role, permission);
+  }
+
+  async checkRolePermission (role: DbRole, permission: string): Promise<boolean | undefined> {
+    const hasPermission = this.hasRolePermission(role.grants, permission);
+    if (hasPermission !== undefined) return hasPermission;
+    else if (role.parentId) {
+      return this.checkParentIdPermission(role.parentId, permission);
+    }
+    return undefined;
+  }
```

Comment:
Is it really a good idea that method can return true, false AND undefined?

Reply:
Yes, it's a bit complicated logic, so i have changed the implementation of PermissionService at all :3